core/app/helpers/spree/taxons_helper.rb

```diff
@@ -4,16 +4,26 @@ module TaxonsHelper
     # that we can use configurations as well as make it easier for end users to override this determination.  One idea is
     # to show the most popular products for a particular taxon (that is an exercise left to the developer.)
     def taxon_preview(taxon, max=4)
-      products = taxon.active_products.select("DISTINCT (spree_products.id), spree_products.*, spree_products_taxons.position").limit(max)
+      products = taxon.active_products.select(*selection_sql).limit(max)
       if (products.size < max)
         products_arel = Spree::Product.arel_table
         taxon.descendants.each do |taxon|
           to_get = max - products.length
-          products += taxon.active_products.select("DISTINCT (spree_products.id), spree_products.*, spree_products_taxons.position").where(products_arel[:id].not_in(products.map(&:id))).limit(to_get)
+          products += taxon.active_products.select(*selection_sql).where(products_arel[:id].not_in(products.map(&:id))).limit(to_get)
           break if products.size >= max
         end
       end
       products
     end
+
+    private
+
+    def selection_sql
+      [
+        "DISTINCT (spree_products.id)",
```

Comment:
Arel does have a node called `Distinct` but I don't think it does anything with the SQL TBH. If anyone can refactor this, that'd be great.

Also, does anyone know why we need to select products by distinct ids? When would there be duplicates?